### PR TITLE
samples: net: https_client: Fix client cert match

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Allow compare to not be strict about ending newline when comparing
the given certificate with the one in the modem.

### CODE REVIEW

**Comment by lemrey** [comment]
We should probably go through the documentation and usage of `modem_key_mgmt_write()` and `modem_key_mgmt_cmp()` and ensure that the intended usage is consistent and properly documented.

What does the documentation say and does it match what the modem expects?
Is the usage aligned across all samples and libraries in NCS?

**Comment by eivindj-nordic** [comment]
> We should probably go through the documentation and usage of `modem_key_mgmt_write()` and `modem_key_mgmt_cmp()` and ensure that the intended usage is consistent and properly documented.
> 
> What does the documentation say and does it match with what the modem expects? Is the usage aligned across all samples and libraries in NCS? 

Usage is aligned in samples and libraries in NCS after this change, this is the only case where one character was subtracted. Modem accepts the certificate both with and without the newline. I've updated the `modem_key_mgmt_cmp` to accept comparing certificates with and without the newline at the end, so it should be a bit more robust now. Was thinking of using `-----END CERTIFICATE-----` string instead, but I don't think that is needed.

**Review by divipillai — APPROVED** [review]

**Review by lemrey — APPROVED** [review]
Could we add a snippet to the library documentation to show how to use the API, so that we can use that as a reference?

**Comment by eivindj-nordic** [comment]
> Could we add a snippet to the library documentation to show how to use the API, so that we can use that as a reference?

Ok, I'll make a separate PR for that.

File: samples/net/https_client/src/main.c [hunk 1]
```diff
@@ -95,11 +95,11 @@ int cert_provision(void)
 		}
 	}
 
-	printk("Provisioning certificate\n");
+	printk("Provisioning certificate to the modem\n");
 
 	/*  Provision certificate to the modem */
 	err = modem_key_mgmt_write(TLS_SEC_TAG, MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN, cert,
-				   sizeof(cert) - 1);
```
Comment (gregersrygg) [inline samples/net/https_client/src/main.c:102]: Can you please explain why this change is needed? Has newline handling changed in `modem_key_mgmt_write`, or was our use incorrect?

Comment (eivindj-nordic) [reply samples/net/https_client/src/main.c:102]: Before this PR, the certificate was written without the last character (`\n` at the end), while in `modem_key_mgmt_cmp()` it is included. So the certificates did not match and it was rewritten. The PR (not removing the last character) aligns with use elsewhere in NCS, while we also add logic to `modem_key_mgmt_cmp()` to make it accept both.

**Review by gregersrygg — APPROVED** [review]
